src/VisualStudio/Core/Def/Implementation/Diagnostics/VisualStudioDiagnosticAnalyzerService.cs

```diff
@@ -199,29 +199,25 @@ public void RunAnalyzers(IVsHierarchy? hierarchy)
             string? projectOrSolutionName = project?.Name ?? PathUtilities.GetFileName(solution.FilePath);
 
             // Add a message to VS status bar that we are running code analysis.
-            var statusMessage = projectOrSolutionName != null
-                ? string.Format(ServicesVSResources.Running_code_analysis_for_0, projectOrSolutionName)
-                : ServicesVSResources.Running_code_analysis_for_Solution;
             var statusBar = _serviceProvider?.GetService(typeof(SVsStatusbar)) as IVsStatusbar;
-            statusBar?.SetText(statusMessage);
+            var totalProjectCount = project != null ? 1 : (uint)solution.ProjectIds.Count;
+            var statusBarUpdater = statusBar != null ?
+                new StatusBarUpdater(statusBar, _threadingContext, projectOrSolutionName, totalProjectCount) :
+                null;
```

Comment:
```suggestion
            var statusBarUpdater = statusBar != null
                ? new StatusBarUpdater(statusBar, _threadingContext, projectOrSolutionName, totalProjectCount)
                : null;
```

Reply:
This matches the existing pattern for conditional expressions used in this file, so I will stick to it.